packages/cli/src/commands/deploy/helper.ts

```diff
@@ -378,6 +381,7 @@ export const createFieldPlugin = async (
     //we need to force an update call right after the creation.
     //If no options is found, it's not going to be sent to the API since undefined
     //properties are not encoded.
+    //NOTE: The `publish` property is set to true here because it is a part of the creation process and should provide a consistent flow.
     await client.updateFieldType({
       id: fieldPlugin.id,
       publish: true,
```

Comment:
I am not 100% happy about this as it is a piece of code that might easily get overlooked, but I think keeping it to publish:true makes sense here, as it kind of mimics the creation process. So I would suggest keeping it to not confuse our users.